api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/DetermineEntitlementMessageProcessor.java

```diff
@@ -75,6 +79,7 @@ public MessageStatus processMessage(Message message) {
                 previousPaymentCycle);
 
         paymentCycleService.updatePaymentCycle(currentPaymentCycle, decision);
+        claimService.updateCurrentIdentityAndEligibilityResponse(claim, decision.getIdentityAndEligibilityResponse());
```

Comment:
I'm thinking that we should maybe refactor these lines into another component, as this message processor has 6 constructor args again
```
        paymentCycleService.updatePaymentCycle(currentPaymentCycle, decision);
        claimService.updateCurrentIdentityAndEligibilityResponse(claim, decision.getIdentityAndEligibilityResponse());
        handleDecision(claim, previousPaymentCycle, currentPaymentCycle, decision, message.getCreatedTimestamp());
        if (claim.getClaimStatus() == PENDING_EXPIRY) {
            paymentCycleService.updateEndDateForClaimBecomingPendingExpiry(currentPaymentCycle);
        }
```
Only problem is, I can't think what to call it. Thoughts?
Either way, it wouldn't be part of this PR

Reply:
I do agree, but bear in mind that one of the dependencies is a config value, so technically its 5 dependencies, so maybe not so desperate. Might be worth a look anyway though.